### ISSUE TITLE
Off-by-one error in date range for low frame counts

### ISSUE THREAD

**Issue by jpatokal** [CartoDB/torque#184]
I have a series of population projections with only 7 timestamps: 2005-1-1, 2010-1-1, ..., 2035-1-1.  I'm loading this into Torque and setting -torque-frame-count: 7.

Expected behavior:
It shows a time slider ranging from 1/2005 to 1/2035, with five steps in between at 1/2010, 1/2015, 1/2020, 1/2025, and 1/2030, corresponding to 30 years (360 months) divided by 6 to get 5 year / 60-month intervals.

Actual behavior:
It shows a time slider ranging from Jan 2005 to Sep 2030, making the 2035 data inaccessible.  Steps are at 4/2009, 7/2013, 11/2017, 2/2022, 6/2026, which corresponds to 30 divided by *7* to get 51.43 month intervals.

This looks like a pretty clear-cut case of [fencepost error](https://en.wikipedia.org/wiki/Off-by-one_error#Fencepost_error
): the number of slider gaps should be computed by counting the spaces between the frames (n-1), not the frames.

Sample: http://cdb.io/1GZekQW

**Comment by fdansv** [comment]
Yes this is something on the backlog, but you've documented it here to perfection, thanks! We'll be tracking it from here.

**Comment by jpatokal** [comment]
Thanks for the confirmation, but looking quickly at the code, is [this line](https://github.com/CartoDB/torque/blob/70800798dab9e93dbcb73c27c396422af05d86b8/lib/torque/provider/windshaft.js#L197
) the culprit?

    this.options.step = (this.options.end - this.options.start)/this.getSteps();

If yes, changing that to `this.getSteps() - 1;` ought to do the trick.  I'll try this out locally if I get a chance in the next few days.
